File: wrappers/java/src/test/java/org/hyperledger/indy/sdk/wallet/CloseWalletTest.java
package org.hyperledger.indy.sdk.wallet;

import org.hyperledger.indy.sdk.IndyIntegrationTest;

import org.junit.Test;

import static org.hamcrest.CoreMatchers.isA;
import static org.junit.Assert.assertNotNull;

import java.util.concurrent.ExecutionException;


public class CloseWalletTest extends IndyIntegrationTest {

	@Test
	public void testCloseWalletWorks() throws Exception {
		Wallet.createWallet(WALLET_CONFIG, WALLET_CREDENTIALS).get();
		Wallet wallet = Wallet.openWallet(WALLET_CONFIG, WALLET_CREDENTIALS).get();

		wallet.closeWallet().get();
	}

	@Test
	public void testAutoCloseWorks() throws Exception {
		Wallet.createWallet(WALLET_CONFIG, WALLET_CREDENTIALS).get();
		try (Wallet wallet = Wallet.openWallet(WALLET_CONFIG, WALLET_CREDENTIALS).get()) {
			assertNotNull(wallet);
		}
		Wallet wallet = Wallet.openWallet(WALLET_CONFIG, WALLET_CREDENTIALS).get();
		wallet.closeWallet().get();
	}
}

File: wrappers/dotnet/indy-sdk-dotnet-test/WalletTests/CloseWalletTest.cs
using Hyperledger.Indy.WalletApi;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Threading.Tasks;

namespace Hyperledger.Indy.Test.WalletTests
{
    [TestClass]
    public class CloseWalletTest : IndyIntegrationTestBase
    {

        [TestMethod]
        public async Task TestCloseWalletWorks()
        {
            await Wallet.CreateWalletAsync(WALLET_CONFIG, WALLET_CREDENTIALS);
            var wallet = await Wallet.OpenWalletAsync(WALLET_CONFIG, WALLET_CREDENTIALS);

            Assert.IsTrue(wallet.IsOpen);

            await wallet.CloseAsync();

            Assert.IsTrue(!wallet.IsOpen);
        }

        [TestMethod]
        public async Task TestCloseWalletWorksForTwice()
        {
            await Wallet.CreateWalletAsync(WALLET_CONFIG, WALLET_CREDENTIALS);
            var wallet = await Wallet.OpenWalletAsync(WALLET_CONFIG, WALLET_CREDENTIALS);

            Assert.IsTrue(wallet.IsOpen);

            await wallet.CloseAsync();

            Assert.IsTrue(!wallet.IsOpen);

            await Assert.ThrowsExceptionAsync<InvalidWalletException>(() =>
                wallet.CloseAsync()
            );
        }

        [TestMethod]
        public async Task TestAutoCloseWorks()
        {
            await Wallet.CreateWalletAsync(WALLET_CONFIG, WALLET_CREDENTIALS);

            using (var a_wallet = await Wallet.OpenWalletAsync(WALLET_CONFIG, WALLET_CREDENTIALS)) {
                Assert.IsNotNull(a_wallet);
                Assert.IsTrue(a_wallet.IsOpen);
            }

            var wallet = await Wallet.OpenWalletAsync(WALLET_CONFIG, WALLET_CREDENTIALS);

            Assert.IsTrue(wallet.IsOpen);

            await wallet.CloseAsync();

            Assert.IsTrue(!wallet.IsOpen);
        }
    }
}
